ariadne/types.py

```diff
@@ -300,6 +302,66 @@ def add_query(self, operation_name: str, query: str):
 """
 QueryParser = Callable[[ContextValue, Dict[str, Any]], DocumentNode]
 
+"""Type of `query_validator` option of GraphQL servers.
+
+Enables customization of server's GraphQL query validation logic. If not set or `None`, 
+default validate is used instead.
+
+# Default query validator
+
+Default query validator used by Ariadne is `validate` function from the `graphql` 
+package.
+
+# Custom validator
+
+Custom validator is a function or callable accepting up to 5 arguments:
+
+`schema`: GraphQLSchema for the graphql schema
+`document_ast`: DocumentNode result of query parser
+`rules`: optional list of AST validation rules of type ASTValidationRule
+`max_errors`: optional maximum number of errors to return
+`type_info`: optional type info, pending deprecation in graphql 3.3
+
+Validator is required to return `List[GraphQLError]` which should be empty
+if there were no errors found
+
+# Example validator
+
+Below code defines custom validator that mutates same Document with a flag to "memoize" validation
+(this works best in a tandem with a memoized parser for end-to-end memoization
+ of parsing and validation):
+
+```python
+from graphql import GraphQLError, validate
+
+
+class MemoizedQueriesValidator:
```

Comment:
Do we need this to be a class? `def memoized_queries_validator()` example would cause less questions to be asked.